### PR TITLE
Prepare to publish to VSCode marketplace

### LINKED ISSUE #42: Release on vscode marketplace
Just wondering if there are any missing features/blockers to be able to push to the market place instead of manually installing from releases :)

### PR DESCRIPTION
**Pull request by noppej**
This PR is to prepare the repo so that we can publish the extension to the VSCode marketplace.

### TODO
- [x] Update the version number in `package.json`
- [x] Add .github/FUNDING.yml for `probe-rs`
- [x] Update the images/probe-rs-debugger `.png` and `.gif`
- [x] Test and update `package` and `publish` scripts in `package.json` - Resolved: Previously blocked on [VSCE issue microsoft/vsmarketplace#585](https://github.com/microsoft/vsmarketplace/issues/585)
- [x] Review and update `README.md`
- [x] CI work
- [x] Review and update instructions and other content on [probe.rs](https://github.com/probe-rs/webpage/pull/71)
- [x] Investigate if/how to release the binary for `probe-rs-debugger` as part of the extension. **Conclusion:** This is a whole new layer of complexity, that belongs in a follow-up PR.

Also See:
- [VSCode Publishing Extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension)

### CODE REVIEW

**Comment by noppej** [comment]
@Yatekii Can you please review, but please DO NOT merge ... I'd like to do that, so that I can test/monitor the publish script.

File: .vscode/extensions.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     // for the documentation about the extensions.json format
     "recommendations": [
         "dbaeumer.vscode-eslint",
```
Comment (Yatekii) [inline .vscode/extensions.json:5]: At least two of those modules have nothing to do with Rust/embedded so I would not recommend them here :)

Comment (Yatekii) [reply .vscode/extensions.json:5]: I would only recommend `hexeditor`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
-## 0.4.2 An updated 'Alpha' release to update dependencies, and improve logging and stability.
-- The majority of functionality is implemented in the [probe-rs-debugger](https://github.com/probe-rs/probe-rs/tree/master/debugger). Please refer to the [probe-rs CHANGELOG](https://github.com/probe-rs/probe-rs/blob/master/CHANGELOG.md) for detailed innformation on changes that affect functionality in this extension.
+## 1.0.0 The first stable release of the extension.
```
Comment (Yatekii) [inline CHANGELOG.md]: I would keep the release number in sync with probe-rs so 0.17.0 atm. Many if not most extensions have 0.x version numbers afaik.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
-## 0.4.2 An updated 'Alpha' release to update dependencies, and improve logging and stability.
-- The majority of functionality is implemented in the [probe-rs-debugger](https://github.com/probe-rs/probe-rs/tree/master/debugger). Please refer to the [probe-rs CHANGELOG](https://github.com/probe-rs/probe-rs/blob/master/CHANGELOG.md) for detailed innformation on changes that affect functionality in this extension.
+## 1.0.0 The first stable release of the extension. 
+- The majority of functionality to support this functionality is implemented in the [probe-rs-debugger](https://github.com/probe-rs/probe-rs/tree/master/debugger).
```
Comment (Yatekii) [inline CHANGELOG.md]:
```suggestion
- The majority of the functionality is implemented in the [probe-rs-debugger](https://github.com/probe-rs/probe-rs/tree/master/debugger).
```

File: README.md [hunk 4]
```diff
@@ -79,15 +71,12 @@ Building the extension refers to the process that generates the installable
...
 
-* Whenever any PR has been merged, you need to create a new release and upload
-  the `.vsix` in the repository's [release
-  page](https://github.com/probe-rs/vscode/releases).
-* This means that every PR needs to update the version number appropriately in
-  the `package.json` file. Please [see the Semantic Versioning
-  guidelines](https://semver.org/).
+Unless you explicitly state otherwise, any contribution intentionally submitted
+for inclusion in the work by you, as defined in the Apache-2.0 license, shall
+be dual licensed as above, without any additional terms or conditions.
```
Comment (Yatekii) [inline README.md]: What dual license above? :)

**Review by Yatekii — COMMENTED** [review]
Overall very nice :) I added a few nits :) Thanks for taking care of this!

File: .vscode/extensions.json (hunk 1 above)
Comment (noppej) [reply .vscode/extensions.json:5]: @Yatekii, these only take effect when you do development on the extension (clone the repo, open the workspace in VSCode), and then you need these. 

For everyday users of the extension, these do not impact them.

File: CHANGELOG.md (hunk 2 above)
Comment (noppej) [reply CHANGELOG.md]: I can live with that :)

File: README.md (hunk 4 above)
Comment (noppej) [reply README.md]: Good catch ... fixed :)

File: .vscode/extensions.json (hunk 1 above)
Comment (Yatekii) [reply .vscode/extensions.json:5]: Ok, sounds good for me!

**Review by Yatekii — APPROVED** [review]

**Pull request merged**